src/components/drawer/drawer.component.js

```diff
@@ -21,20 +22,65 @@ const Drawer = ({
   animationDuration,
   ...props
 }) => {
+  const drawerSidebarContentRef = useRef();
   const isControlled = expanded !== undefined;
   const [isExpandedInternal, setIsExpandedInternal] = useState(defaultExpanded || false);
-  const [isFirstOpen, setFirstOpen] = useState(true);
+  const [isOpening, setIsOpening] = useState(false);
+  const [isClosing, setIsClosing] = useState(false);
   const isExpanded = isControlled ? expanded : isExpandedInternal;
+  const timer = useRef();
+
+  const getAnimationDuration = useCallback(() => {
+    if (animationDuration.indexOf('ms') !== -1) {
+      const animationTime = animationDuration.substring(0, animationDuration.length - 2);
+      return animationTime;
+    }
+
+    if (animationDuration.indexOf('.') !== -1) {
+      const animationTime = animationDuration.substring(0, animationDuration.length - 1);
+      return parseFloat(animationTime) * 1000;
+    }
+
+    return animationDuration;
+  }, [animationDuration]);
+
+  const toggleAnimation = useCallback(() => {
+    const timeout = getAnimationDuration();
+    clearTimeout(timer);
+    if (!isExpanded) {
+      setIsOpening(true);
+      timer.current = setTimeout(() => {
+        setIsOpening(false);
+      }, timeout);
+    } else {
+      setIsClosing(true);
+      timer.current = setTimeout(() => {
+        setIsClosing(false);
+      }, timeout);
+    }
+  }, [timer, getAnimationDuration, isExpanded]);
```

Comment:
timer is a ref, so no need for it in dependencies

Reply:
fixed